component/loki/source/windowsevent/component_windows.go

```diff
@@ -0,0 +1,148 @@
+package windowsevent
+
+import (
+	"context"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/common/loki"
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/grafana/loki/clients/pkg/promtail/scrapeconfig"
+	"github.com/grafana/loki/clients/pkg/promtail/targets/windows"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name: "loki.source.windowsevent",
+		Args: Arguments{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+var (
+	_ component.Component = (*Component)(nil)
+)
+
+// Component implements the loki.source.windowsevent component.
+type Component struct {
+	opts component.Options
+
+	mut       sync.RWMutex
+	args      Arguments
+	target    *windows.Target
+	handler   chan api.Entry
+	receivers []loki.LogsReceiver
+}
+
+func (c *Component) Chan() chan<- api.Entry {
+	return c.handler
+}
+
+func (c *Component) Stop() {
+	// This is a noop.
+}
+
+// New creates a new loki.source.windowsevent component.
+func New(o component.Options, args Arguments) (*Component, error) {
+
+	c := &Component{
+		opts:      o,
+		receivers: args.ForwardTo,
+		handler:   make(chan api.Entry),
+		args:      args,
+	}
+
+	// Call to Update() to start readers and set receivers once at the start.
+	if err := c.Update(args); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+// Run implements component.Component.
+func (c *Component) Run(ctx context.Context) error {
+	defer func() {
+		c.mut.Lock()
+		defer c.mut.Unlock()
+		if c.target != nil {
+			_ = c.target.Stop()
+		}
+	}()
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case entry := <-c.handler:
+			lokiEntry := loki.Entry{
+				Labels: entry.Labels,
+				Entry:  entry.Entry,
+			}
+			for _, receiver := range c.receivers {
+				receiver <- lokiEntry
+			}
```

Comment:
Do we need to take an RLock/RUnlock for c.receivers? Also, can we send the entry directly?

Reply:
Good call on lock, and we cant because it's a slightly different struct with almost the exact same fields.